src/Raven.Server/Documents/Queries/AST/TimeSeriesBetweenExpression.cs

```diff
@@ -0,0 +1,50 @@
+namespace Raven.Server.Documents.Queries.AST
+{
+    public class TimeSeriesBetweenExpression : BetweenExpression
+    {
+        public QueryExpression MinExpression => Min ?? _minExpression;
+        public QueryExpression MaxExpression => Max ?? _maxExpression;
+
+        private readonly QueryExpression _minExpression;
+        private readonly QueryExpression _maxExpression;
+
+        public TimeSeriesBetweenExpression(QueryExpression source, QueryExpression min, QueryExpression max) : base(source, null, null)
+        {
+            if (min is ValueExpression ve)
+            {
+                Min = ve;
+            }
+            else
+            {
+                _minExpression = min;
+            }
```

Comment:
This looks really awkward, why do we have to this it this way?

Reply:
removed this